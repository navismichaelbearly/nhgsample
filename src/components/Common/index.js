import styled from 'styled-components';
import { Button, mediaQueries, Typography } from 'nhh-styles';

/**
 * file to aggregate styled components and hopefully allow more reuse of them
 * also to eventually replace all the labels used as heading etc
 */
export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;

export const NotificationWrapper = styled.div`
  margin-bottom: 30px;
`;
