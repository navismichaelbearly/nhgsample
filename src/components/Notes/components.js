import styled from 'styled-components';
import { Typography } from 'nhh-styles';

export const NoteItem = styled.div`
  margin-bottom: 1rem;
`;

export const NoteBody = styled.div`
  word-break: break-all;
`;

export const NoteLabel = styled.span`
  font-weight: 700;
`;

export const NoteDescription = styled.p`
  margin-top: 1rem;
`;

export const Wrapper = styled.div`
  clear: both;
  margin-bottom: 1.5rem;
  padding-top: 1rem;
`;

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;
