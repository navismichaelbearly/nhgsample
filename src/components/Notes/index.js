import { Note, NotePropType, NoteDictionaryPropType } from './Note';

export { default as NoteList } from './NoteList';
export { Note, NotePropType, NoteDictionaryPropType };
