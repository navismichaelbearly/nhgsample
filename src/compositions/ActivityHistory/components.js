import styled from 'styled-components';
import { Button, Typography } from 'nhh-styles';

export const Activity = styled.div`
  margin-bottom: 1rem;
`;

export const ActivityBody = styled.div``;

export const ActivityLabel = styled.span`
  font-weight: 700;
`;

export const ActivityLink = styled(Button)`
  font-size: inherit;
`;

export const H4 = styled(Typography.H4)`
  font-size: 1em;
  margin-bottom: 0.7rem;
`;

export const Wrapper = styled.div`
  clear: both;
  margin-bottom: 1.5rem;
  padding-top: 1rem;
`;

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;

export const ActivityDescription = styled.span`
  word-break: break-word;
`;
