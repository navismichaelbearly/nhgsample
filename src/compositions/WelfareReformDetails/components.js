import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Row = styled.div`
  margin-bottom: 15px;
`;

export const FootNote = styled.div`
  border: 1px;
  color: #aaa;
  font-size: 14px;
`;
