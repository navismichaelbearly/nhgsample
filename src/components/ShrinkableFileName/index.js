import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'nhh-styles';
import styled from 'styled-components';
import { shortenFileName } from '../../util/splitAndWrap';
import { LabelWrapper } from './components';

const shorterNameForDesktop = shortenFileName(26);
const shorterNameForMobile = shortenFileName(7);

const AttachmentsTableFileName = ({ name, isLink, onClick }) => {
  const ParentElement = isLink ? Button : styled.span``;
  const parentProps = isLink ? { isText: true, onClick } : {};

  return (
    <LabelWrapper>
      <ParentElement {...parentProps}>
        <span className="d-none d-lg-block">{shorterNameForDesktop(name)}</span>
        <span className="d-lg-none">{shorterNameForMobile(name)}</span>
      </ParentElement>
    </LabelWrapper>
  );
};

AttachmentsTableFileName.defaultProps = {
  isLink: false,
  onClick: () => {},
};

AttachmentsTableFileName.propTypes = {
  name: PropTypes.string.isRequired,
  isLink: PropTypes.bool,
  onClick: PropTypes.func,
};

export default AttachmentsTableFileName;
