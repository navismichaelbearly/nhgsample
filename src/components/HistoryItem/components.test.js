import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Label, Value, Title, Description } from './components';

describe('HistoryItem components', () => {
  it('should render <Label /> correctly', () => {
    expect(shallow(<Label />)).toMatchSnapshot();
  });
  it('should render <Value /> correctly', () => {
    expect(shallow(<Value />)).toMatchSnapshot();
  });
  it('should render <Title /> correctly', () => {
    expect(shallow(<Title />)).toMatchSnapshot();
  });
  it('should render <Description /> correctly', () => {
    expect(shallow(<Description />)).toMatchSnapshot();
  });
});
