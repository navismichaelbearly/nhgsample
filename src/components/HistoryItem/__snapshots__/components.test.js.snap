// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem components should render <Description /> correctly 1`] = `
.c0 {
  overflow: hidden;
  padding: 1rem;
}

<div
  className="c0"
/>
`;

exports[`HistoryItem components should render <Label /> correctly 1`] = `
.c0 {
  font-weight: 700;
}

<span
  className="c0"
/>
`;

exports[`HistoryItem components should render <Title /> correctly 1`] = `
.c0 {
  margin: -30px 0 30px 0;
}

<styled.h3
  className="c0"
/>
`;

exports[`HistoryItem components should render <Value /> correctly 1`] = `
.c0 {
  word-break: break-word;
}

<span
  className="c0"
/>
`;
